smoke-tests/src/test/java/com/splunk/opentelemetry/TelemetryRetriever.java

```diff
@@ -0,0 +1,118 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+
+class TelemetryRetriever {
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  private final OkHttpClient client;
+  private final int backendPort;
+
+  TelemetryRetriever(OkHttpClient client, int backendPort) {
+    this.client = client;
+    this.backendPort = backendPort;
+  }
+
+  void clearTelemetry() throws IOException {
+    client
+        .newCall(
+            new Request.Builder()
+                .url(String.format("http://localhost:%d/clear", backendPort))
+                .build())
+        .execute()
+        .close();
+  }
+
+  TraceInspector waitForTraces() throws IOException, InterruptedException {
+    Stream<JsonNode> content = waitForContent("get-traces");
+
+    return new TraceInspector(
+        content.map(this::deserializeTraceRequest).collect(Collectors.toList()));
+  }
+
+  MetricsInspector waitForMetrics() throws IOException, InterruptedException {
+    Stream<JsonNode> content = waitForContent("get-metrics");
+
+    return new MetricsInspector(
+        content.map(this::deserializeMetricsRequest).collect(Collectors.toList()));
+  }
+
+  private ExportTraceServiceRequest deserializeTraceRequest(JsonNode it) {
+    var builder = ExportTraceServiceRequest.newBuilder();
+    deserializeIntoBuilder(it, builder);
+    return builder.build();
+  }
+
+  private ExportMetricsServiceRequest deserializeMetricsRequest(JsonNode it) {
+    var builder = ExportMetricsServiceRequest.newBuilder();
+    deserializeIntoBuilder(it, builder);
+    return builder.build();
+  }
+
+  private void deserializeIntoBuilder(JsonNode it, GeneratedMessageV3.Builder<?> builder) {
+    // TODO(anuraaga): Register parser into object mapper to avoid de -> re ->
```

Comment:
I would remove the todo or at least the name.  😉

Reply:
I'll remove the entire TODO: I've just spent a few minutes looking at this and I don't think that there's a one-liner that fixes this problem. And writing a custom deserializer seems like an overkill -- converting back to string may not be pretty but it's the simplest way.